cql3/statements/modification_statement.hh

```diff
@@ -42,7 +42,7 @@ namespace raw { class modification_statement; }
 class modification_statement : public cql_statement_opt_metadata {
 public:
     const statement_type type;
-
+    std::vector<uint16_t> partition_key_bind_indexes;
```

Comment:
Since we don't really need all indexes, we could just store a flag, e.g. `_may_use_token_aware_routing`.

Reply:
I need someone more familiar with CQL to review whether looking at partition_key_bind_indexes is really including all the cases of token awareness. @avikivity who would that be?

Reply:
I guess @sylwiaszunejko by looking at what drivers actually do for token aware routing. I can't imagine it's anything other than building a key using partition_key_bind_indexes and calculating the token from it.

Reply:
It is exactly like that on the driver side

Reply:
> It is exactly like that on the driver side

Ok.